### DISCUSSION TITLE
Umarshal the batch of spans coming from otel collector in form of protobuf

### DISCUSSION THREAD

**Discussion by scarshad** [open-telemetry/opentelemetry-java-contrib#1094] [Q&A]
I need to unmarshal the batch of spans that are received from the collector. [This is the format.](https://github.com/open-telemetry/opentelemetry-proto/blob/main/examples/trace.json) Should I copy paste the trace.proto and use this or you guys using something else to extract the spans?
In go, there is exposed library which do these tasks and we can extract the data by calling methods, not able to find similar methods in Java too
go example
```go
                         rs := spans.ScopeSpans().At(j).Spans()
			scopeName := spans.ScopeSpans().At(j).Scope().Name()
			scopeVersion := spans.ScopeSpans().At(j).Scope().Version()
			for k := 0; k < rs.Len(); k++ {
				r := rs.At(k)
				spanAttr := attributesToMap(r.Attributes())
				status := r.Status()
				eventTimes, eventNames, eventAttrs := convertEvents(r.Events())
				linksTraceIDs, linksSpanIDs, linksTraceStates, linksAttrs := convertLinks(r.Links())
```

**Comment by jkwatson** [comment]
There are published bindings here you might be able to leverage: https://github.com/open-telemetry/opentelemetry-proto-java

**Reply by scarshad** [reply]
Thanks @jkwatson will try above library for unmarshaling the spans which are coming from collector and update in this discussion.
